kcidb/test_db.py

```diff
@@ -19,194 +19,77 @@ def test_schemas_main():
-            status = function()
-        Client.assert_called_once_with("bigquery:project.dataset")
-        client.cleanup.assert_called_once()
-        return status
-    """)
-    assert_executes("", *argv, driver_source=driver_source)
+    clean_database.init()
+    argv = ["kcidb.db.cleanup_main", "-d", clean_database.database]
+    assert_executes("", *argv)
 
 
-def test_empty_main():
+def test_empty_main(empty_database):
     """Check kcidb-db-empty works"""
-    argv = ["kcidb.db.empty_main", "-d", "bigquery:project.dataset"]
-    driver_source = textwrap.dedent("""
-        from unittest.mock import patch, Mock
-        client = Mock()
-        client.empty = Mock()
-        client.is_initialized = Mock(return_value=True)
-        with patch("kcidb.db.Client", return_value=client) as \
-                Client:
-            status = function()
-        Client.assert_called_once_with("bigquery:project.dataset")
-        client.empty.assert_called_once()
-        return status
-    """)
-    assert_executes("", *argv, driver_source=driver_source)
 
+    argv = ["kcidb.db.empty_main", "-d", empty_database.database]
+    assert_executes("", *argv)
 
-def test_dump_main():
+# review
+def test_dump_main(clean_database):
     """Check kcidb-db-dump works"""
+
     empty = kcidb.io.SCHEMA.new()
-    argv = ["kcidb.db.dump_main", "-d", "bigquery:project.dataset",
+    clean_database.is_initialized()
+    argv = ["kcidb.db.dump_main", "-d", clean_database.database,
             "--indent=0"]
 
-    driver_source = textwrap.dedent(f"""
-        from unittest.mock import patch, Mock
-        client = Mock()
-        client.dump_iter = Mock(return_value=iter(({repr(empty)},)))
-        with patch("kcidb.db.Client", return_value=client) as \
-                Client:
-            status = function()
-        Client.assert_called_once_with("bigquery:project.dataset")
-        client.dump_iter.assert_called_once()
-        return status
-    """)
-    assert_executes("", *argv, driver_source=driver_source,
-                    stdout_re=re.escape(json.dumps(empty) + "\n"))
-
-    driver_source = textwrap.dedent(f"""
-        from unittest.mock import patch, Mock
-        client = Mock()
-        client.dump_iter = Mock(return_value=iter({repr((empty, empty))}))
-        with patch("kcidb.db.Client", return_value=client) as \
-                Client:
-            status = function()
-        Client.assert_called_once_with("bigquery:project.dataset")
-        client.dump_iter.assert_called_once()
-        return status
-    """)
-    assert_executes("", *argv, driver_source=driver_source,
-                    stdout_re=re.escape(json.dumps(empty) + "\n" +
-                                        json.dumps(empty) + "\n"))
+    assert_executes("", *argv, stdout_re=re.escape(json.dumps(empty) + "\n"))
+    assert_executes("", *argv, stdout_re=re.escape(json.dumps(empty) + "\n" +
+                                                   json.dumps(empty) + "\n"))
```

Comment:
This would only work before, because we mocked the client returning the data. We don't need this exact test anymore. The one I describe above would be better.